src/FacilitatorOptionParser.ts

```diff
@@ -0,0 +1,179 @@
+      onlyFacilitator = true;
+      configObj = this.handleFacilitatorConfigOption();
+    }
+
+    if (onlyFacilitator === false) {
+      this.isOriginAuxChainDefined();
+      configObj = this.handleOriginAuxChainOption();
+    }
+
+    return configObj;
+  }
+
+  /**
+   * This method returns Config object when origin chain and aux chain is defined.
+   * @returns Config object encapsulating facilitator and mosaic configs.
+   */
+  private handleOriginAuxChainOption(): Config {
+    // When facilitator config is provided.
+    if (this.facilitatorConfigPath) {
+      const facilitatorConfig: FacilitatorConfig = FacilitatorConfig.fromPath(
+        this.facilitatorConfigPath,
+      );
+
+      this.verifyChainIdInFacilitatorConfig(facilitatorConfig);
+
+      // when mosaic config path is given.
+      if (this.mosaicConfigPath) {
+        const mosaicConfig: MosaicConfig = MosaicConfig.fromFile(this.mosaicConfigPath);
+        // verify origin chain and aux chain is present in mosaic config.
+        this.verifyChainIdInMosaicConfig(mosaicConfig);
+        return Config.getConfigFromPath(this.mosaicConfigPath, this.facilitatorConfigPath);
+      }
+
+      const mosaicConfig: MosaicConfig = MosaicConfig.fromChain(this.originChain);
+      return new Config(mosaicConfig, facilitatorConfig);
+    }
+
+    if (this.mosaicConfigPath) {
+      const mosaic: MosaicConfig = MosaicConfig.fromFile(this.mosaicConfigPath);
+      this.verifyChainIdInMosaicConfig(mosaic);
+      const facilitator = FacilitatorConfig.from(this.auxChain);
+      return new Config(mosaic, facilitator);
+    }
+
+    const facilitator: FacilitatorConfig = FacilitatorConfig.from(this.auxChain);
+    const mosaic: MosaicConfig = MosaicConfig.fromChain(this.originChain);
+    return new Config(mosaic, facilitator);
+  }
+
+  /**
+   * This method returns config object when facilitator config is provided and
+   * origin chain and aux chain is not provided.
+   * @returns Config object encapsulating facilitator and mosaic configs.
+   */
+  private handleFacilitatorConfigOption(): Config {
+    let configObj = {} as any as Config;
+    // When no origin and aux chain provided.
+    if (this.mosaicConfigPath) {
+      configObj = Config.getConfigFromPath(this.mosaicConfigPath, this.facilitatorConfigPath);
+    } else {
+      const facilitatorConfig: FacilitatorConfig = FacilitatorConfig.fromPath(
+        this.facilitatorConfigPath,
+      );
+      const mosaicConfig: MosaicConfig = MosaicConfig.fromChain(
+        facilitatorConfig.originChainId,
+      );
+      configObj = new Config(mosaicConfig, facilitatorConfig);
+    }
+    return configObj;
+  }
+
+  /**
+   * It verifies chain ids in facilitator config.
+   * @param facilitatorConfig Facilitator object containing facilitator config.
+   */
+  private verifyChainIdInFacilitatorConfig(
+    facilitatorConfig: FacilitatorConfig,
+  ): void {
+    if (facilitatorConfig.chains[this.auxChain] === undefined
```

Comment:
Why do we check undefined, null? I am trying to understand the logic. Should it be just `!== ''` ?

Reply:
`chains` is an `Record` in facilitatorConfig object. It would return `undefined` if the `auxchain` is present.